tests/unit/metadata/bucketfile/backend.js

```diff
@@ -0,0 +1,73 @@
+const EventEmitter = require('events');
+const assert = require('assert');
+
+const extensions = require('arsenal').algorithms.list;
+
+const BucketFileInterface =
+    require('../../../../lib/metadata/bucketfile/backend').default;
+
+const KEY_LENGTH = 15;
+const KEY_COUNT = 1000;
+const MAX_KEYS = 100;
+
+const KEY_TEMPLATE = '0'.repeat(KEY_LENGTH + 1);
+
+function zpad(key) {
+    return `${KEY_TEMPLATE}${key}`.slice(-KEY_LENGTH);
+}
+
+class Reader extends EventEmitter {
+    constructor() {
+        super();
+        this.done = false;
+        this.index = 0;
+        process.nextTick(() => this.start());
+    }
+
+    start() {
+        if (this.done) {
+            return null;
+        }
+        const i = this.index++;
+        // extensions should take into account maxKeys
+        // and should not filter more than that intented value
+        assert(i <= MAX_KEYS, `listed more than maxKeys ${MAX_KEYS}`);
+        if (i === KEY_COUNT) {
+            return this.emit('end');
+        }
+        this.emit('data', { key: `${zpad(i)}`,
+            value: `{"foo":"${i}","initiator":"${i}"}` });
+        return process.nextTick(() => this.start());
+    }
+
+    destroy() {
+        this.done = true;
+    }
+}
+
+describe('BucketFileInterface::internalListObject', alldone => {
+    const bucketfile = new BucketFileInterface();
+
+    // stub db to inspect the extensions
+    const db = {
+        createReadStream: () => new Reader(),
+    };
+
+    // stub functions and components
+    const logger = { info: () => {}, debug: () => {}, error: () => {} };
+    // eslint-disable-next-line no-unused-vars
+    const loadDBIfExists = bucketfile.loadDBIfExists;
```

Comment:
why is this defined?  To restore the function after the test?  If so, let's do that now in an after block.